### PR TITLE
Aliases and groups

### PR DESCRIPTION
**Pull request by MrTravisB**
We're working on adding support for groups and aliases to our API. For groups our `identify` endpoints just takes a `group_id` and `group_attributes` params. For aliases the `identify` endpoint takes a `previous_id` param.

Please let me know if this implemented right.

Please don't merge this yet. We'll won't be releasing these features until later this week. Just wanted to get this out there in case it was wrong so that it can be ready to go when we are ready to deploy.

### CODE REVIEW

**Comment by stevenmiller888** [comment]
@ndhoule @f2prateek @amillet89 Can you review this?

File: lib/index.js [hunk 1]
```diff
@@ -70,3 +70,22 @@ Outbound.prototype.group = function(payload, fn){
...
+* @param {Alias} alias
+* @param {Function} fn
+* @api private
+*/
+
+Outbound.prototype.alias = function(payload, fn){
+
+  // Outbound v2
+  return this
+  .post('/v2/identify')
```
Comment (f2prateek) [inline lib/index.js:86]: nit: 2 spaces on this line and for each below.

File: lib/mapper.js [hunk 2]
```diff
@@ -68,8 +68,22 @@ exports.group = function(group){
...
+* Map alias.
+*
+* @param {Alias} alias
+* @return {Object}
+* @api private
+*/
+
+exports.alias = function(alias) {
+    return {
+        user_id: group.userId(),
```
Comment (f2prateek) [inline lib/mapper.js]: Shouldn't `group` be `alias` here?

File: lib/mapper.js [hunk 3]
```diff
@@ -68,8 +68,22 @@ exports.group = function(group){
...
+        previous_id: group.previousId()
```
Comment (f2prateek) [inline lib/mapper.js]: Same as above.

**Comment by f2prateek** [comment]
Few nits inline, lgtm otherwise.

Would still need tests before merging whenever you go live.

**Comment by MrTravisB** [comment]
@f2prateek I've updated with the changes you requested.

We have deployed everything on our end. You are free to merge and deploy whenever you are ready.

**Comment by zachkinloch** [comment]
This is Zach from DoorDash, we're eagerly waiting the for this to be merged and deployed before the weekend. Critical for us to get the alias feature ASAP!  Thanks.

**Comment by f2prateek** [comment]
@MrTravisB This is still missing tests for the alias method. Also I'm not sue if you've run the existing tests local or not, but pretty sure the `group` tests will fail since you updated the mapping but not the fixtures.

**Comment by MrTravisB** [comment]
@f2prateek I updated all of the tests. Had to fix all the code style errors in the tests just to run them which is why I had to change the track and identify helpers.

**Comment by f2prateek** [comment]
@MrTravisB sorry for the existing errors, and thanks for fixing those!

**Pull request closed without merging**